Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -1753,21 +1753,26 @@ Normalize(std::vector<double> & x)
 static bool
 IsAffine(const mat44 & nifti_mat)
 {
-  const vnl_matrix_fixed<float, 4, 4> mat(&(nifti_mat.m[0][0]));
+  vnl_matrix_fixed<double, 4, 4> mat;
+
+  for (int i = 0; i < 4; ++i)
+    for (int j = 0; j < 4; ++j)
+      mat[i][j] = double{ nifti_mat.m[i][j] };
+
   // First make sure the bottom row meets the condition that it is (0, 0, 0, 1)
   {
-    float bottom_row_error = itk::Math::abs(mat[3][3] - 1.0f);
+    double bottom_row_error = std::fabs(mat[3][3] - 1.0);
     for (int i = 0; i < 3; ++i)
     {
-      bottom_row_error += itk::Math::abs(mat[3][i]);
+      bottom_row_error += std::fabs(mat[3][i]);
     }
     if (bottom_row_error > std::numeric_limits<float>::epsilon())
     {
       return false;
     }
   }
 
-  const float condition = vnl_matrix_inverse<float>(mat.as_matrix()).well_condition();
+  const double condition = vnl_matrix_inverse<double>(mat.as_matrix()).well_condition();
   // Check matrix is invertible by testing condition number of inverse
   if (!(condition > std::numeric_limits<float>::epsilon()))
```

Comment:
Should `std::numeric_limits<float>` here be replaced with `std::numeric_limits<double>`? I'm just asking, I do not know it myself.